core/services/llo/telemetry.go

```diff
@@ -39,7 +39,11 @@ func NewTelemeterService(lggr logger.Logger, monitoringEndpoint commontypes.Moni
 }
 
 func newTelemeter(lggr logger.Logger, monitoringEndpoint commontypes.MonitoringEndpoint, donID uint32) *telemeter {
-	chTelemetryObservation := make(chan TelemetryObservation, 100)
+	// NOTE: This channel must take multiple telemetry packets per round (1 per
+	// feed) so we need to make sure the buffer is large enough.
+	//
+	// 2000 feeds * 5s/250ms = 40_000 should hold ~5s of buffer in the worst case.
+	chTelemetryObservation := make(chan TelemetryObservation, 40_000)
```

Comment:
should we make the buffer length increase with the concurrency setting? (eg TransmitConcurrency * 200)?

Reply:
I don't think transmit concurrency affects it directly. We might consider making it configurable but I'd honestly prefer to choose a large enough conservative value that we don't have to worry about it, the only real penalty is using some extra memory